components/tasks/TaskTable.tsx

```diff
@@ -1,222 +0,0 @@
-import { components } from '@/lib/types/api';
-import { Button } from '@chakra-ui/button';
-import { Box, Flex, Heading } from '@chakra-ui/layout';
-import {
-    FormControl,
-    FormErrorMessage,
-    FormLabel,
-    Input,
-    Modal,
-    ModalCloseButton,
-    ModalContent,
-    ModalFooter,
-    ModalHeader,
-    ModalOverlay,
-    useDisclosure,
-} from '@chakra-ui/react';
-import { Table, Tbody, Td, Th, Thead, Tr } from '@chakra-ui/table';
-import React, { useState } from 'react';
-import { useSWRConfig } from 'swr';
-import { FormStatus } from '../projects/NewProject/reducer';
-import * as fetch from '@/lib/utils/fetch';
-import ErrorAlert from '../common/ErrorAlert';
-import useData from '@/lib/hooks/useData';
-import Loading from '../common/Loading';
-import checkDateOrder from '@/lib/utils/checkDateOrder';
-
-type TaskTableProps = {
-    project?: components['schemas']['ProjectDTO'];
-};
-
-type StateType = {
-    name: string;
-    description: string;
-    startDate: string;
-    endDate?: string;
-    formStatus: FormStatus;
-    errorMessage: string;
-};
-
-function TaskTable({ project }: TaskTableProps): JSX.Element {
-    const { isOpen, onOpen, onClose } = useDisclosure();
-    const [state, setState] = useState<StateType>({
-        name: '',
-        description: '',
-        startDate: new Date().toISOString().split('T')[0],
-        formStatus: FormStatus.IDLE,
-        errorMessage: '',
-    });
-
-    const {
-        data: tasks,
-        isError: taskError,
-        isLoading: tasksLoading,
-    } = useData<components['schemas']['TaskDTO'][]>('task', { projectId: `${project?.id}` });
-
-    const url = `${process.env.NEXT_PUBLIC_API_URL}/task`;
-    const { mutate } = useSWRConfig();
-
-    const handleSubmit = async (e: React.MouseEvent) => {
-        e.preventDefault();
-
-        const createTaskRequest: components['schemas']['TaskDTO'] = {
-            name: state.name,
-            description: state.description,
-            startDate: state.startDate,
-            endDate: state.endDate,
-            project: project,
-        };
-        setState({
-            ...state,
-            formStatus: FormStatus.LOADING,
```

Comment:
I don't understand why we need this `formStatus` thing. I'm just going to suggest we remove it throughout and render ´<Error>` components iff `errorMessage` is defined.